File: benchmarks/numerical/matrix_100.js
const N = 100;
const A = Array.from({length: N}, (_, i) => Array.from({length: N}, (_, j) => (i*7+j*13) % 97));
const B = Array.from({length: N}, (_, i) => Array.from({length: N}, (_, j) => (i*11+j*17) % 89));
const start = Date.now();
const C = Array.from({length: N}, (_, i) => Array.from({length: N}, (_, j) => {
    let sum = 0;
    for (let k = 0; k < N; k++) sum += A[i][k] * B[k][j];
    return sum;
}));
console.log((Date.now() - start) * 1000);

File: benchmarks/numerical/matrix_100.rs
use std::time::Instant;
fn main() {
    const N: usize = 100;
    let mut a = [[0.0f64; N]; N];
    let mut b = [[0.0f64; N]; N];
    let mut c = [[0.0f64; N]; N];
    for i in 0..N {
        for j in 0..N {
            a[i][j] = ((i*7 + j*13) % 97) as f64;
            b[i][j] = ((i*11 + j*17) % 89) as f64;
        }
    }
    let start = Instant::now();
    for i in 0..N {
        for j in 0..N {
            let mut sum = 0.0;
            for k in 0..N { sum += a[i][k] * b[k][j]; }
            c[i][j] = sum;
        }
    }
    println!("{}", start.elapsed().as_micros());
}
